tools/scxtop/src/main.rs

```diff
@@ -62,6 +71,177 @@ fn get_action(_app: &App, keymap: &KeyMap, event: Event) -> Action {
     }
 }
 
+/// Attaches BPF programs to the skel.
+fn attach_progs(skel: &mut BpfSkel) -> Result<Vec<Link>> {
+    // Attach probes
+    let mut links = vec![
+        skel.progs.on_sched_cpu_perf.attach()?,
+        skel.progs.scx_sched_reg.attach()?,
+        skel.progs.scx_sched_unreg.attach()?,
+        skel.progs.on_sched_switch.attach()?,
+        skel.progs.on_sched_wakeup.attach()?,
+        skel.progs.on_sched_wakeup_new.attach()?,
+    ];
+
+    // 6.13 compatibility
+    if let Ok(link) = skel.progs.scx_insert_vtime.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dispatch_vtime.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_insert.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dispatch.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dispatch_from_dsq_set_vtime.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dsq_move_set_vtime.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dsq_move_set_slice.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dispatch_from_dsq_set_slice.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dispatch_from_dsq.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.scx_dsq_move.attach() {
+        links.push(link);
+    }
+    if let Ok(link) = skel.progs.on_cpuhp_enter.attach() {
+        links.push(link);
+    }
+    if compat::ksym_exists("gpu_memory_total").is_ok() {
+        if let Ok(link) = skel.progs.on_gpu_memory_total.attach() {
+            links.push(link);
+        }
+    }
+
+    Ok(links)
+}
+
+fn run_trace(trace_args: &TraceArgs) -> Result<()> {
+    TermLogger::init(
+        match trace_args.verbose {
+            0 => simplelog::LevelFilter::Info,
+            1 => simplelog::LevelFilter::Debug,
+            _ => simplelog::LevelFilter::Trace,
+        },
+        SimplelogConfig::default(),
+        TerminalMode::Mixed,
+        ColorChoice::Auto,
+    )?;
+
+    let config = Config::default_config();
+    let worker_threads = config.worker_threads() as usize;
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .worker_threads(if worker_threads > 2 {
+            worker_threads
+        } else {
+            4
+        })
+        .build()
+        .unwrap()
+        .block_on(async {
+            let (action_tx, mut action_rx) = mpsc::unbounded_channel();
+
+            let mut open_object = MaybeUninit::uninit();
+            let mut builder = BpfSkelBuilder::default();
+            if trace_args.verbose > 2 {
+                builder.obj_builder.debug(true);
+            }
+
+            let skel = builder.open(&mut open_object)?;
+            // set sample rate to 1 here to populate the BPF tctxs
+            skel.maps.data_data.sample_rate = 1;
```

Comment:
nit: could you set mode to MODE_TRACING here? It doesn't do anything yet but would be neater.